### ISSUE TITLE
Make getKeyByXXX hydration safe (for Next.js)

### ISSUE THREAD

**Issue by QzCurious** [ecyrbe/zodios#243]
## What happened

I want hydrate react-query cache with Next.js. By since data to be hydrated should be json serializable.

A snapshot demonstrate non-serializable error on Next.js (caused by a object property with `undefined` value):

![image](https://user-images.githubusercontent.com/38932402/203462231-85dd43a3-fff7-4212-87e5-eb08d7211401.png)

### Remove property with value `undefined`

Here is a great [code snippet](https://www.30secondsofcode.org/js/s/compact-object) to remove falsy value from object or array recursively. But I'm currently not able to typescript it.

### Change  implementation of `pick` util

Turns out that `getKeyByXXX` use internal [`pick`](https://github.com/ecyrbe/zodios-react/blob/884b295d3ded6c674d857a5aa5315ef4d83ab704/src/utils.ts#L18-L35) util function to extract keys out of config. It accidentially includes new keys into the returned object. Maybe it's more correct to make `pick` into

```diff
export function pick<T, K extends keyof T>(
  obj: T | undefined,
  keys: K[]
): Pick<T, K> {
  const ret = {} as Pick<T, K>;
  if (obj) {
    for (const key of keys) {
+      if (key in obj) {
        ret[key] = obj[key];
+      }
    }
  }
  return ret;
}
```

But not sure if this change would cause existing code break at runtime.

## Sidenote

react-query is not stopping us from making `undefined` queryKey (like `[{ name: undefined }]`).
> Some storage systems (such as browser [Web Storage API](https://developer.mozilla.org/en-US/docs/Web/API/Web_Storage_API)) require values to be JSON serializable. If you need to dehydrate values that are not automatically serializable to JSON (like Error or undefined), you have to serialize them for yourself.
>  [react-query doc](https://tanstack.com/query/v4/docs/reference/hydration#limitations)

**Comment by ecyrbe** [comment]
Hello,

Thank you for taking the time to debug and find a solution to this issue.
A patch is coming.

**Comment by ecyrbe** [comment]
this is now fixed in `@zodios/react` `v10.3.4`
